[DOCS] DOC-298 Data Assistant technical term page

Rachel-Reverie opened the pull request:
Changes proposed in this pull request:
- [x] Adds a technical term page for Data Assistants
- [x] Adds a technical term tag for Data Assistants

### Definition of Done
- [ ] I have made corresponding changes to the documentation

Rachel-Reverie committed:
DOC-298
- first draft of the technical term page

Rachel-Reverie committed:
Merge branch 'develop' into d/doc-298/data_assistant_technical_term_page

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=6057&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=6057&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/6057/8720ee7f/c969d473393ac27a7766bbe42fd4335fca285f7c.svg)](https://app.codesee.io/r/reviews?pr=6057&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

Rachel-Reverie committed:
DOC-298
- update link to reflect changes to guide in Develop branch
- Add technical term tag entry to terms.json
- Add "Data Assistant" entry to glossary of terms and ToC.

Rachel-Reverie committed:
DOC-298
- update definition per feedback recieved

Rachel-Reverie committed:
DOC-298
- removes emdash from definition; replaces with comma

Rachel-Reverie committed:
DOC-298
- attempt to remove "special characters" from prior cut'n'paste of definition

Rachel-Reverie committed:
DOC-298
- correct mismatch in one of the locations the term is defined.

Rachel-Reverie committed:
DOC-298
- re-enter Data Assistants tag data to ensure no remaining issues from previous cut'n'paste containing "strange characters"

Rachel-Reverie committed:
DOC-298
- correct capitalization of technical term

Rachel-Reverie committed:
DOC-298
- updates definition according to additional guidance.

Rachel-Reverie committed:
DOC-298
- added technical term tags

alexsherstinsky commented on docs/glossary.md:
```diff
@@ -19,6 +19,8 @@ title: "Glossary of Terms"
 
 [**Data Asset:**](./terms/data_asset.md) A collection of records within a Datasource which is usually named based on the underlying data system and sliced to correspond to a desired specification.
 
+[**Data Assistant:**](./terms/data_assistant.md) A utility that asks questions about your data, gathering information to describe what is observed, and then presents Metrics and proposes Expectations based on the answers.
```
@Rachel-Reverie I would add to the definition something like "A goal-specific utility" (whereby goal could be Onboarding, Volume, etc.  Thanks!

Rachel-Reverie replied:
Updated in the next commit.  Thank you!

alexsherstinsky commented on docs/term_tags/terms.json:
```diff
@@ -39,6 +39,11 @@
   },
+  "data_assistant": {
+    "term": "Data Assistant",
+    "definition": "A utility that asks questions about your data, gathering information to describe what is observed, and then presents Metrics and proposes Expectations based on the answers.",
```
@Rachel-Reverie Some suggestion as above.  Thanks.

alexsherstinsky commented on docs/terms/data_assistant.md:
```diff
@@ -0,0 +1,81 @@
+
+### Definition
+
+A Data Assistant is a utility that asks questions about your data, gathering information to describe what is observed, and then presents <TechnicalTag tag="metric" text="Metrics" /> and proposes <TechnicalTag tag="expectation" text="Expectations" /> based on the answers.
```
@Rachel-Reverie Some suggestion as above.  Thanks.

alexsherstinsky commented on docs/terms/data_assistant.md:
```diff
@@ -0,0 +1,81 @@
+
+### Easy profiling
+
+Data Assistants can function as pre-configured Profilers.  Since you can access them directly from your Data Context, they are easily accessible.  This ensures that they will always provide a quick, simple entry point to creating Expectations and <TechnicalTag tag="profiling" text="Profiling" /> your data.  However, the rules implemented by a Data Assistant are fully exposed in the parameters for ...
```
@Rachel-Reverie In fact, a Data Assistant is exactly a pre-configured Rule-Based Profiler.  Specifically, the Data Assistant is put together by writing rules that together fit the theme of the given Data Assistant goal at hand.  Happy to discuss/elaborate.  Thanks!

Rachel-Reverie replied:
Yes, but all previous versions of profilers have, to the best of my knowledge, been exclusively oriented toward creating an Expectation Suite.  Since we're differentiating from the past behavior (Data Assistants can be used to generate Metrics and plots without creating an Expectation Suite) it seemed it would be more intuitive for our users to say that they "can" be used as a pre-configured Profiler (according to the previous understanding of what a Profiler does and produces).  But they also have additional functionality.

Perhaps it would be better to say: "Data Assistants are preconfigured Profilers with extended versatility." ?

alexsherstinsky commented on docs/terms/data_assistant.md:
```diff
@@ -0,0 +1,81 @@
+
+### Multi-Batch introspection
+
+Data Assistants really show their worth when you provide them a Batch Request that contains multiple Batches.  With previous Profilers you would only be able to introspect a single Batch at a time.  This meant that the Expectation Suite generated would only reflect a single Batch.  If you had many Batches of data that you wanted to build inter-related Expectations for, you would have needed to ru ...
```
@Rachel-Reverie This is a bit more nuanced -- happy to discuss to offer a perspective.  Thanks!

alexsherstinsky commented on docs/terms/data_assistant.md:
```diff
@@ -0,0 +1,81 @@
+
+### Design motivation
+
+Data Assistants were designed to make creating Expectations easier for users of Great Expectations.  A Data Assistant will help solve the problem of "where to start" when working with a large, new, or complex dataset by greedily asking questions and building a list of all the relevant Expectations that it can determine from the answers to those questions.  Branching question paths ensure that add ...
```
@Rachel-Reverie Strictly speaking, `Expectations` is not the only output (e.g., Metrics as well, or could be only Metrics; but not sure if/how we would want to communicate this fact).  Thanks!

Rachel-Reverie replied:
Perhaps: "The result is a comprehensive gathering of Metrics that can be saved, reviewed as graphical plots, or used by the Data Assistant to generate a set of proposed Expectations." ?

alexsherstinsky commented on docs/terms/data_assistant.md:
```diff
@@ -0,0 +1,81 @@
+
+### Additional documentation
+
+Data Assistants are multi-batch aware out of the box.  However, not every use case requires multiple Batches.  For more information on when it is best to work with either a single Batch or multiple Batches of data in a Batch Request, please see the following guide:
```
@Rachel-Reverie As mentioned above, this is a little nuanced -- happy to discuss.  Thanks!

Rachel-Reverie committed:
DOC-298
- updated definition per https://github.com/great-expectations/great_expectations/pull/6057#discussion_r976926445

Rachel-Reverie committed:
DOC-298
- updated definition in glossary and terms.json

Rachel-Reverie committed:
DOC-298
- updates "Easy Profiling" heading per https://github.com/great-expectations/great_expectations/pull/6057#discussion_r976930131

Rachel-Reverie committed:
DOC-298
- updates "Design motivation" heading per https://github.com/great-expectations/great_expectations/pull/6057#discussion_r976931732

Rachel-Reverie committed:
DOC-298
- updates "Multi-Batch introspection" section per conversation post https://github.com/great-expectations/great_expectations/pull/6057#discussion_r976930871

Rachel-Reverie committed:
Merge branch 'develop' into d/doc-298/data_assistant_technical_term_page
